src/video/cocoa/SDL_cocoawindow.m

```diff
@@ -2020,8 +2020,15 @@ - (BOOL)acceptsFirstMouse:(NSEvent *)theEvent
      */
     SDL_WindowData *data = (SDL_WindowData *) window->driverdata;
     Cocoa_WindowListener *listener = data->listener;
+    NSWindow *nswindow = data->nswindow;
     if (![listener isInFullscreenSpace]) {
         SetWindowStyle(window, GetWindowStyle(window));
+        if (resizable) {
+            /* resizable windows are Spaces-friendly: they get the "go fullscreen" toggle button on their titlebar. */
+            [nswindow setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
```

Comment:
`NSWindowCollectionBehaviorFullScreenPrimary` is for 10.7+, yes?
How will this behave on 10.6?

Reply:
There is an ifdef guard here that specifies that it only works on 10.7 SDK or above: https://github.com/libsdl-org/SDL/blob/565ef63873c2dfa85e1e80db9dfbefbd9a8c67db/src/video/cocoa/SDL_cocoawindow.m#L25

Reply:
Ah, there is some if statements that check videodata->allow_spaces. I'll add that here.

Reply:
I meant at run time, not at compile time.

Reply:
I made a force push that will check `videodata->allow_spaces` at runtime which has the 10.6 check.

Reply:
I see, OK.